.github/workflows/test-build-and-deploy.yml

```
@@ -129,11 +131,12 @@ jobs:
           path: coverage
       - name: Display structure of downloaded files
         run: ls -R
-      - uses: codecov/codecov-action@v3
+      - uses: codecov/codecov-action@v4.0.1
```

```suggestion
      - uses: codecov/codecov-action@v4
```
Github recommends using major version for actions